### PR TITLE
Documentation on performance enhancement changes

### PR DESCRIPTION
**Pull request by lcampos**
### What does this PR do?
Documentation on how to enable the deploy/retrieve performance enhancements.

### What issues does this PR fix or reference?
@W-6531270@

### CODE REVIEW

File: docs/_articles/en/user-guide/perf-enhancements.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: Performance enhancements
+lang: en
+---
+
+The performance enhancement changes allows you to run single file metadata deploys through a more efficient code path. These changes currently only support Apex Class metadata types. More metadata types will get included in the following releases.
```
Comment (sfsholden) [inline docs/_articles/en/user-guide/perf-enhancements.md]:
```suggestion
The performance enhancement changes allow you to run single file metadata deploys through a more efficient code path. These changes currently only support Apex Class metadata types. Additional metadata types will be supported in future releases.
```

File: docs/_articles/en/user-guide/perf-enhancements.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+> NOTICE: Performance enhancement changes for single Apex Class file deploys is currently in beta. If you find any bugs or have feedback, [open a GitHub issue](./en/bugs-and-feedback).
+
+## Setup
+
+Because the performance enhancement changes are in beta, you must enable them:
```
Comment (sfsholden) [inline docs/_articles/en/user-guide/perf-enhancements.md]:
```suggestion
To enable this beta feature:
```

File: docs/_articles/en/user-guide/perf-enhancements.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+1. Select **File** > **Preferences** > **Settings** (Windows or Linux) or **Code** > **Preferences** > **Settings** (macOS).
+1. Under Salesforce Feature Previews, select Experimental: Deploy Retrieve.
+
+In this beta release, we have released performance enhancement changes when running the **SFDX: Deploy this Source to Org** command only when running it for Apex Class.
```
Comment (sfsholden) [inline docs/_articles/en/user-guide/perf-enhancements.md]:
```suggestion
In this beta release, performance enhancements will only occur when running the **SFDX: Deploy this Source to Org**  command from an Apex Class.
```

Comment (jag-sfdc) [reply docs/_articles/en/user-guide/perf-enhancements.md]: How about? "In this beta release, the performance enhancements are only enabled for **SFDX: Deploy this Source to Org**  command from an Apex Class."

File: docs/_articles/ja/user-guide/perf-enhancements.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+title: Performance enhancements
+lang: ja
+---
+
+The performance enhancement changes allows you to run single file metadata deploys through a more efficient code path. These changes currently only support Apex Class metadata types. More metadata types will get included in the following releases.
```
Comment (AnanyaJha) [inline docs/_articles/ja/user-guide/perf-enhancements.md]: I think we need the same changes here as were made above ^^

File: docs/_articles/en/user-guide/perf-enhancements.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+> NOTICE: Performance enhancement changes for single Apex Class file deploys is currently in beta. If you find any bugs or have feedback, [open a GitHub issue](./en/bugs-and-feedback).
+
+## Setup
+
+To enable this beta feature:
+
+1. Select **File** > **Preferences** > **Settings** (Windows or Linux) or **Code** > **Preferences** > **Settings** (macOS).
+1. Under Salesforce Feature Previews, select Experimental: Deploy Retrieve.
+
+In this beta release, performance enhancements will only occur when running the **SFDX: Deploy this Source to Org** command from an Apex Class.
```
Comment (AnanyaJha) [inline docs/_articles/en/user-guide/perf-enhancements.md]: Should we specify which code paths are affected? ie. the right click menus and the command palette invokations?

File: docs/_articles/ja/user-guide/perf-enhancements.md (hunk 4 above)
Comment (sfsholden) [reply docs/_articles/ja/user-guide/perf-enhancements.md]: Good catch

File: docs/_articles/en/user-guide/perf-enhancements.md (hunk 5 above)
Comment (lcampos) [reply docs/_articles/en/user-guide/perf-enhancements.md]: In both code paths the command name is `SFDX: Deploy this Source to Org`, I think we can add it or leave it as is. Whatever you think makes it clearer.

**Pull request merged**